python/ray/serve/tests/test_metrics.py

```diff
@@ -1107,13 +1107,23 @@ async def hang_on_first_request():
         timeout=15,
         metric="ray_serve_num_scheduling_tasks",
         expected=-1,  # -1 means not expected to be present yet.
+        # TODO(zcin): this tag shouldn't be necessary, there shouldn't be a mix of
+        # metrics from new and old sessions.
+        expected_tags={
+            "SessionName": ray._private.worker.global_worker.node.session_name
+        },
```

Comment:
what's the context for this change? not seeing how it's relevant to this PR

Reply:
I ran into a weird issue where the metric `ray_serve_num_scheduling_tasks` (among a few other metrics that are recorded on the router) was still present when the ray cluster was shutdown and restarted, just tagged with the old cluster session name. According to @rkooo567 this is a bug. It's unclear why this bug was triggered by my changes in this PR, but it shouldn't be a bug in Serve (e.g. the router isn't being created when it shouldn't be, and there aren't any false metrics being reported for the new cluster). This is a workaround to get the test to pass for now.

Reply:
Please create an issue!